### ISSUE TITLE
Option to download at a specified rate (to avoid exceeding query rate)

### ISSUE THREAD

**Issue by chiarnglin** [althonos/InstaLooter#212]
Feature request: Option to download at a specified rate (to avoid exceeding query rate)

Description: Currently, IG has a rate limit. For users with a large number of posts (>2000), it's almost guaranteed to hit the `Query rate exceeded` limit before you can download all the posts. One way around this is to set a limit to the query rate from the client-side, so that we don't hit the limit. We can then leave instalooter running in Terminal overnight, and all posts will be downloaded. 

Acceptance Criteria:
- new Files Option `-r NUM` or `--rate-limit NUM`, which specifies the number of files per minute to attempt downloading. 
- complete downloading all media from profiles with large number of posts (>2000) without hitting `Query rate exceeded` limit by setting a very low NUM.

**Comment by distantstar** [comment]
Well, you can already do this by modifying the interval variable to greater than 2 seconds in pages.py.

**Comment by chiarnglin** [comment]
Any user with some programming background can fork their own repo and maintain it, but that's the purpose of creating a feature request: 1) so everyone can use it as a feature and 2) nobody has to spend time rebasing on every new instalooter version release. 

If it's such a simple change, it should be trivial to add as a feature.

**Comment by jpegcoma** [comment]
Yup would be great to have this functionality. In batch mode, I keep getting query rate error with login error in some cases

**Comment by andy-br** [comment]
Even with changing the query rate in pages.py it still hits the limit at around 1000 posts. Any ideas for a solution?
